docs/spec/data-plane.md

```diff
@@ -0,0 +1,76 @@
+# Knative Eventing Data Plane Contracts
+
+## Introduction
+
+Developers using Knative Eventing need to know what is supported for delivery to user provided components that receive events. Knative Eventing defines contract for data plane components and we have listed them here.
+
+## Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC2119.
+
+## Data plane contract for Sinks
+
+A **Sink** is an [_addressable_](./interfaces.md#addressable) resource that takes responsibility for the event. A Sink could be a consumer of events, or middleware. A Sink MUST be able to receive CloudEvents over HTTP and HTTPS.
+
+A **Sink** MAY be [_callable_](./interfaces.md#callable) resource that represents an Addressable endpoint which receives an event as input and optionally returns an event to forward downstream
+
+Almost every component in Knative Eventing may be a Sink providing composability.
+
+Every Sink MUST support HTTP Protocol Binding for CloudEvents [version 1.0](https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md) and [version 0.3](https://github.com/cloudevents/spec/blob/v0.3/http-transport-binding.md) with restrictions and extensions specified below.
+
+### HTTP Support
+
+This section adds restrictions on [requirements in HTTP Protocol Binding for CloudEvents](https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md#12-relation-to-http).
+
+Sinks MUST accept HTTP requests with POST method and MAY support other HTTP methods.
+If a method other is not supported Sink SHOULD respond with HTTP status code `405 Method Not Supported`.
+Non-event requests (e.g. health checks) are not constrained.
+
+The URL used by a Sink MUST correspond to a single, unique
+endpoint at any given moment in time. This MAY be done via the host, path, query
+string, or any combination of these. This mapping is handled exclusively by the
+[Addressable control-plane](./interfaces.md#control-plane) exposed via the `status.address.url`.
```

Comment:
<!--mdfmt-bot-->
Format markdown:
```suggestion
The URL used by a Sink MUST correspond to a single, unique endpoint at any given
moment in time. This MAY be done via the host, path, query string, or any
combination of these. This mapping is handled exclusively by the
[Addressable control-plane](./interfaces.md#control-plane) exposed via the
`status.address.url`.
```